packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/query/QueryPredicateBuilder.kt

```diff
@@ -58,7 +58,12 @@ class QueryPredicateBuilder {
                     }
                 }
 
-                val queryField: QueryField = QueryField.field(field)
+                // Here we are using query root model name to create QueryField to let amplify-android
+                // generate correct SQL command.
+                // This is based on the current assumption: amplify-flutter doesn't support cross models nested
+                // predicate e.g. query comments by post.id
```

Comment:
I thought this was possible. `where: Comment.POST.eq(postID)`. Are you referring to something else?

Reply:
this comment is for

```dart
await Amplify.DataStore.query(
  Comment.classType,
  where: Post.ID.eq('123),
)
```

(This is currently supported only in amplify-android)